test/model/src/driver/debug_log_parser.cpp

```diff
@@ -644,13 +722,33 @@ debug_log_parser::from_json(kv_database &database, const char *path)
             continue;
         }
 
+        /* Handle the relevant system entries. */
+        if (log_entry_type == "system") {
+
+            /* Replay all supported system operations. */
+            for (auto &op_entry : log_entry.at("ops")) {
+                std::string op_type = op_entry.at("optype").get<std::string>();
+
+                /* Previous LSN. */
+                if (op_type == "prev_lsn") {
+                    parser.apply(op_entry.get<prev_lsn>());
+                    continue;
+                }
+            }
+
+            continue;
+        }
+
         /* Ignore these fields. */
         if (log_entry_type == "checkpoint" || log_entry_type == "file_sync" ||
-          log_entry_type == "message" || log_entry_type == "system")
+          log_entry_type == "message")
             continue;
 
         throw model_exception("Unsupported log entry type \"" + log_entry_type + "\"");
     }
+
+    /* Simulate the database starting up. */
+    database.start();
```

Comment:
New line 458 and new line 751 both startup the database. Why are both needed? Could both ever be called for the same database? Perhaps comments giving the use case are needed on both of them.

Reply:
I clarified this in the comments, thank you! Line 458 handles database restarts recorded in the log. Line 751 handles the startup that would happen after the model is populated from the debug logs, when the database is open for verification.